packages/backend/helpers/crm/hubspot.ts

```diff
@@ -72,6 +74,132 @@ export const getHubspotAssociationObj = (key: AllAssociation, associateObj: Stan
+
+                //collecting all ids for batch request.
+                const ids = (associatedData[objectType] as any[]).map((item) => item.id);
+
+                if (ids.length > 0) {
+                    let index = 0;
+                    let fullBatchData: any[] = [];
+
+                    while (index < ids.length) {
+                        const batchIds = ids.slice(index, index + 100); //the batch api only takes 100 at a time
+
+                        //bacth request for that object type to get full data for that object.
+                        const responseData = await axios({
+                            method: 'post',
+                            url: `https://api.hubapi.com/crm/v3/objects/${objectType}/batch/read`,
+                            headers: {
+                                authorization: `Bearer ${thirdPartyToken}`,
+                                'content-type': 'application/json',
+                            },
+                            data: JSON.stringify({
+                                inputs: batchIds,
+                            }),
+                        });
+                        index += 100;
+
+                        fullBatchData = [...fullBatchData, ...responseData.data.results];
+
+                        if (responseData.headers['x-hubspot-ratelimit-secondly-remaining'] === 0) {
+                            await new Promise((resolve) => setTimeout(resolve, 1000));
+                        }
+                    }
+
+                    //converting the objectType into the correct objType which is needed for unification.
+                    const associatedObjectType = getStandardObjects(objectType as PluralObjectType);
+                    // if the objectType is among the objects we support,we unify it,else let it be as it is.
+
+                    associatedData[objectType] =
+                        associatedObjectType && associatedObjectType !== null
+                            ? await Promise.all(
+                                  fullBatchData.map(
+                                      async (item: any) =>
+                                          await unifyObject<any, any>({
+                                              obj: {
+                                                  ...item,
+                                                  ...item?.properties,
+                                              },
+                                              tpId: thirdPartyId,
+                                              objType: associatedObjectType,
+                                              tenantSchemaMappingId: connection.schema_mapping_id,
+                                              accountFieldMappingConfig: account.accountFieldMappingConfig,
+                                          })
+                                  )
+                              )
+                            : fullBatchData;
+                }
+            }
+        }
+    }
+
```

Comment:
@manish-singh-bisht currently it looks like if objectType isn't supported by us, we are returning as it is without unifiying. instead what we want is in the response, we want the warning but api call should not fail for rest, ie: something like an error object which includes what all associations type which we don't support, have an message for those if you want thissupport contact us.

also what happens when for eg: we pass `associations=` or `associations=undefined` or something similar, etc, are we handling those cases ?

Reply:
@manish-singh-bisht makes sense user won't use but what if they pass `associations=` or anything similar, that will make a call to hubspot right ? + also can you check what hubspot api responds with when we pass anything random or something similar that might crash things, ain't it ?

Reply:
1) what if the user wants the associated data for an object we don't support?

Reply:
as suggested by jatin, we should give warning about it that we don't support this object in response somewhere, note we shouldn't fail the call + don't make api call to hubspot for that association.

Reply:
2) it will work for associatons= ,but for associations=undefined it will use "undefined" as the the value of associations parameter resulting in a error.

btw why will the user do this associations=undefined?

Reply:
> @manish-singh-bisht makes sense user won't use but what if they pass `associations=` or anything similar, that will make a call to hubspot right ? + also can you check what hubspot api responds with when we pass anything random or something similar that might crash things, ain't it ?

when we pass something to associations that is not recognized by hubspot it will throw error

Reply:
@manish-singh-bisht cool, so inorder to not crash (500 error). don't make the call if there is undefined/null/or empty string for that association to hubspot itself + in response let users know someway that, 
`no such association exist in hubspot or if that exist and we are not supporting contact us`

Reply:
@Nabhag8848  now no calls will be there to hubspot other than the objects we support, for every other object it will send an error message (this will not stop the flow of the call for the other supported objects).

undefined and null will not crash.

empty too will not throw error